android/src/main/java/com/swmansion/rnscreens/bottomsheet/SheetDetents.kt

```
@@ -0,0 +1,73 @@
+package com.swmansion.rnscreens.bottomsheet
+
+import com.swmansion.rnscreens.Screen
+
+class SheetDetents(
+    private val rawDetents: List<Double>,
+) {
+    init {
+        require(rawDetents.isNotEmpty()) { "[$TAG] At least one detent must be provided." }
+        require(rawDetents.size <= 3) { "[$TAG] Maximum of 3 detents supported." }
+        rawDetents.forEach {
+            require(it in 0.0..1.0 || it == SHEET_FIT_TO_CONTENTS) {
+                "[$TAG] Detent values must be within 0.0 and 1.0, or SHEET_FIT_TO_CONTENTS should be defined, got $it."
+            }
+        }
+    }
+
+    val count: Int get() = rawDetents.size
+
+    fun empty(): Boolean = count == 0
+
+    fun at(index: Int): Double = rawDetents[index]
+
+    fun last(): Double = rawDetents.last()
+
+    fun heightAt(
+        index: Int,
+        containerHeight: Int,
+    ): Int = (at(index).coerceIn(0.0, 1.0) * containerHeight).toInt()
```

do we need `coerceIn` if we have `require(it in 0.0..1.0 || ...)` in `init`? What about `SHEET_FIT_TO_CONTENTS` (`-1.0`), should we silently convert this to `0.0` or should we require that detent is not `SHEET_FIT_TO_CONTENTS` when `heightAt` is called (or throw if it is)?